packages/shared/friends/sagas.ts

```diff
@@ -1410,6 +1411,61 @@ export async function getChannelMessages(request: GetChannelMessagesPayload) {
   getUnityInstance().AddChatMessages(addChatMessages)
 }
 
+// Search channels
+export async function searchChannels(request: GetChannelsPayload) {
+  const client: SocialAPI | null = getSocialClient(store.getState())
+  if (!client) return
+
+  const searchTerm: string = request.name
+
+  // get user joined channelIds
+  const joinedChannelIds = getAllConversationsWithMessages(store.getState())
+    .filter((conv) => conv.conversation.type === ConversationType.CHANNEL)
+    .map((conv) => conv.conversation.id)
+
+  // get user profile
+  const profile = getCurrentUserProfile(store.getState())
+
+  // search channels
+  const { channels, nextBatch } = await client.searchChannel(searchTerm, request.limit, request.since)
+
+  // parse channel info
+  const channelsToReturn: ChannelInfoPayload[] = []
+
+  for (const channel of channels) {
+    let joined = false
+    let muted = false
+
+    if (joinedChannelIds.includes(channel.id)) {
+      joined = true
+    }
+    if (profile?.muted?.includes(channel.id)) {
+      muted = true
+    }
+    channelsToReturn.push({
+      name: channel.name!,
+      channelId: channel.id,
+      unseenMessages: 0,
+      lastMessageTimestamp: undefined,
+      memberCount: channel.memberCount,
+      description: channel.description || '',
+      joined,
+      muted
+    })
+  }
```

Comment:
nit: I think we can simplify this mapping, like:

```suggestion
  const channelsToReturn = channels.map((channel) => ({
    channelId: channel.id,
    name: channel.name || '',
    unseenMessages: 0,
    lastMessageTimestamp: undefined,
    memberCount: channel.memberCount,
    description: channel.description || '',
    joined: joinedChannelIds.includes(channel.id),
    muted: profile?.muted?.includes(channel.id) || false
  }))
```